addons/im_livechat/static/src/core/web/livechat_conversation_tag_edit.js

```diff
@@ -0,0 +1,121 @@
+    static components = {};
+    static props = ["thread", "autofocus?", "close?", "selectedTags?"];
+    static template = "im_livechat.ConversationTagEdit";
+
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        this.store = useService("mail.store");
+        this.inputRef = useAutofocus();
+        this.sequential = useSequential();
+        this.state = useState({
+            selectableTags: [],
+            selectedTags: [...(this.props.selectedTags || [])],
+            searchStr: "",
+        });
+        this.debouncedFetchConversationTags = useDebounced(
+            this.fetchConversationTags.bind(this),
+            250
+        );
+        onWillStart(() => {
+            this.fetchConversationTags();
+        });
+        useEffect(
+            () => {
+                this.debouncedFetchConversationTags();
+            },
+            () => [this.state.searchStr]
+        );
+        useEffect(
+            () => {
+                this.sequential(async () => {
+                    const selectedTagIds = this.state.selectedTags.map((tag) => tag.id).sort();
+                    await this.orm.write("discuss.channel", [this.props.thread.id], {
+                        livechat_conversation_tag_ids: selectedTagIds.length
+                            ? selectedTagIds
+                            : false,
+                    });
+                });
+            },
+            () => [this.state.selectedTags]
+        );
+    }
+
+    get allSelectableTagNames() {
+        return this.state.selectableTags.map((tag) => tag.name);
+    }
+
+    get allSelectedTagNames() {
+        return this.state.selectedTags.map((tag) => tag.name);
+    }
+
+    get remainingSelectableTags() {
+        return this.state.selectableTags.filter((tag) => !tag.in(this.state.selectedTags));
+    }
+
+    toggleSelectedTag(tag) {
+        if (this.state.selectedTags.includes(tag)) {
+            this.state.selectedTags = this.state.selectedTags.filter((t) => t !== tag);
+        } else {
+            this.state.selectedTags = [...this.state.selectedTags, tag];
+        }
+    }
+
+    async fetchConversationTags() {
+        const results = await this.sequential(() =>
+            this.orm.searchRead(
+                "im_livechat.conversation.tag",
+                [["name", "ilike", this.state.searchStr]],
+                ["id", "name"],
+                { limit: 15 }
+            )
+        );
+        if (!results) {
+            return;
+        }
+        const result = this.store["im_livechat.conversation.tag"].insert(results);
+        this.state.selectableTags = [...result];
+    }
+
+    onKeydownSearchInput(ev) {
+        if (ev.key === "Enter") {
+            ev.preventDefault();
+            this.onClickCreateToggle();
+        }
+    }
+
+    onClickFooterSelectedTag(tag) {
+        this.state.selectedTags = this.state.selectedTags.filter((t) => t !== tag);
```

Comment:
To fix the reactivity of the list to bus changes, we should keep a stack of "changes" that are done here instead, and re-apply them on the result.

So the end result would be "current thread values" + all added tags - all removed tags.